source/Octopus.Tentacle/Kubernetes/KubernetesVersionParser.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Text.RegularExpressions;
+using k8s.Models;
+
+namespace Octopus.Tentacle.Kubernetes
+{
+    public static class KubernetesVersionParser
+    {
+        public static ClusterVersion ParseClusterVersion(VersionInfo versionInfo)
+        {
+            return new ClusterVersion(SanitizeAndParseVersionNumber(versionInfo.Major), SanitizeAndParseVersionNumber(versionInfo.Minor));
+        }
+        
+        static int SanitizeAndParseVersionNumber(string version)
+        {
+            if (int.TryParse(version, out var result))
+                return result;
+            
+            var sanitized = Regex.Replace(version, "[^0-9]", "");
```

Comment:
Is there a reason we don't just compile the Regex and always sanitise, rather than trying to parse, then using the regex if it can't?

Reply:
There isn't a proper reason 😂 , I just went with the method that co-pilot generated initially. Ill update it to always sanitize since I think that will be more readable and I doubt we get any performance gains by doing the `TryParse` first anyway

Reply:
@liam-mackie I had that thought but this check is only run once (as it's cached in a `AsyncLazy`), so the optimization of a compiled regex isn't that necessary. We probably don't need the `TryParse`, but I don't mind it as it's the expected result

Reply:
@APErebus I was just curious (for my own purposes) - I assumed that we didn't run the Regex constantly because there was a performance concern if we parsed regardless, so wondered why we'd do the `TryParse` instead of a precompiled Regex (I.E. is a precompiled regex still terrible to use 😅 )